### ISSUE TITLE
roslaunch gives deceptive error if chmod +x isn't set

### ISSUE THREAD

**Issue by xkortex** [ros/ros_comm#1532]
Trying to `roslaunch pack_name launcher.launch` fails with the following error if your source file exists in the appropriate spot, but does not have `+x` permission set. 

```
ERROR: cannot launch node of type [pack_name/node_name.py]: 
can't locate node [node_name.py] in package [pack_name]
```

Looks like it is from: 

https://github.com/ros/ros_comm/blob/eb7e686a9a58109087c956eed3cfa6f2ed3eec3d/tools/roslaunch/src/roslaunch/node_args.py#L256

I spent a lot of time mucking around with path, ENV, packages.xml, namespaces, and the launch file, thinking it is a problem with a path or naming scheme. Once you know what you are looking for, it is obvious from questions on the wiki that this is a [commonly](https://answers.ros.org/question/244463/errorcannot-launch-node-of-typetesttestcannot-locate-node-test-in-package-test/) [encountered](https://answers.ros.org/question/209426/i-cant-launch-a-node/) [problem](https://answers.ros.org/question/241690/roslaunch-error-error-cannot-launch-node-of-type/). 

If the file is `+x` but lacks a shebang, the more helpful error occurs:

```
Unable to launch [foo/image_nexus_node-2]. 
If it is a script, you may be missing a '#!' declaration at the top.
```
Could someone add a more useful helper text here to indicate that this might be a permission error and not a path error? Perhaps: 

```
raise NodeParamsException("can't locate node [%s] in package [%s]. 
Make sure file exists in package path and permission is set to executable (chmod +x)"
%(node.type, node.package))
```

**Comment by dirk-thomas** [comment]
> Could someone add a more useful helper text here to indicate that this might be a permission error and not a path error?

Please consider to contribute a pull request with your proposal.

**Comment by dirk-thomas** [comment]
Addressed by #1568.
